### DISCUSSION TITLE
Required init with backing field gives null warning

### DISCUSSION THREAD

**Discussion by JoasE** [dotnet/csharplang#6752] [Q&A] [answered]
Having a required init property which sets a backing field still gives a null warning.
Is this a bug?

The below code gives a warning:  Warning	CS8618	Non-nullable field '_name' must contain a non-null value when exiting constructor. Consider declaring the field as nullable.

```csharp
public class TestRequiredInit
{
    private readonly string _name;

    public required string Name
    {
        get => _name;
        init => _name = value;
    }
}
```

![image](https://user-images.githubusercontent.com/32096708/204270405-d94e87cc-80e3-436e-bf31-3d1682137ce8.png)

I don't see a way to create `TestRequiredInit` without `_name` being set to a non nullable value without producing warnings, se the warning it produces now doesn't make sense to me.

Doing this with constructor instead of required init doesn't produce a warning.
```csharp
public class TestConstructor
{
    private readonly string _name;

    public TestConstructor(string name)
    {
        _name = name;
    }

    public string Name
    {
        get => _name;
    }
}
```
![image](https://user-images.githubusercontent.com/32096708/204270462-0fa6867f-5cdc-4f4a-9a73-b82b1b0c1cd1.png)

**Comment by 333fred** [comment, marked as answer]
As it currently stands, this behavior is expected. However, we will be discussing a possible improvement to the scenario in LDM on Wednesday: https://github.com/dotnet/csharplang/issues/6754.

**Reply by 333fred** [reply]
Note for posterity: LDM approved making a change here, `MemberNotNullAttribute` on `Name` can be used in this scenario.
